src/builder.rs

```diff
@@ -111,6 +111,72 @@ impl<W: Write> Builder<W> {
         append(self.get_mut(), header, &mut data)
     }
 
+    /// Appends the contents of an existing archive to this archive.
+    ///
+    /// This function will iterate through the entries of the supplied archive,
+    /// appending them one by one to the target.
+    ///
+    /// # Errors
+    ///
+    /// This function will return an error for any intermittent I/O error which
+    /// occurs when either reading or writing.
+    ///
+    /// # Examples
+    ///
+    /// ```no_run
+    /// use std::fs::File;
+    /// use tar::{Archive, Builder};
+    ///
+    /// let mut input = Archive::new(File::open("/path/to/existing.tar").unwrap());
+    ///
+    /// let mut output_file = File::create("/path/to/target.tar").unwrap();
+    /// let mut output = Builder::new(output_file);
+    ///
+    /// output.append_archive(&mut input).unwrap();
+    /// output.finish().unwrap();
+    /// ```
+    pub fn append_archive<R: Read>(&mut self, archive: &mut Archive<R>) -> io::Result<()> {
+        let entries = archive.entries()?.raw(true);
+
+        for entry in entries {
+            self.append_entry(entry?)?;
+        }
+
+        Ok(())
+    }
+
+    /// Appends an entry from an existing archive to this archive.
+    ///
+    /// This function will append an existing `Entry` to the builder target.
```

Comment:
Could some caveats be mentioned here about things like PAX metadata and I think long symlink names not being preserved just yet?

Reply:
I'm struggling to replicate GNU `@LongLink` or PAX metadata loss in a test. I've added some additional test resources - `pax3.tar` and `gnu-longname.tar` - to try and push these edge cases (see [append_entry_from_existing_archive](https://github.com/alexcrichton/tar-rs/blob/0fe050da0a6702373b06d8192393ee2393f91187/tests/all.rs#L403)). 

Any ideas about how I can enhance these tests to catch the unsupported cases?